src/main/java/io/beanmapper/core/converter/impl/OptionalToObjectConverter.java

```diff
@@ -1,18 +1,64 @@
 package io.beanmapper.core.converter.impl;
 
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 
 import io.beanmapper.BeanMapper;
 import io.beanmapper.core.BeanPropertyMatch;
 import io.beanmapper.core.converter.BeanConverter;
+import io.beanmapper.utils.Classes;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This converter facilitates the conversion of an arbitrary amount of Optional wrappers, however, support for complex
+ * datastructures, such as Maps, Sets, List, etc. is limited to a single layer. As such, if the user requires support
+ * deeper layers of complex datastructures within Optionals, they should revise their system design, and if that does
+ * not solve the problem, create a custom converters for their specific conversion.
+ */
 public class OptionalToObjectConverter implements BeanConverter {
 
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public <S, T> T convert(BeanMapper beanMapper, S source, Class<T> targetClass, BeanPropertyMatch beanPropertyMatch) {
+        if (targetClass.equals(Optional.class) && logger.isDebugEnabled())
```

Comment:
Klopt het dat je hier eigenlijk zegt dat als de target een Optional is en de logging staat op debug dat je dan pas converten? Dat zou betekenen dat als debug uit staat je geen optionals kan mappen.

Reply:
Nee, er wordt alleen gelogged wanneer debugging enabled is, maar zal dat sowieso even refactoren voor de duidelijkheid.